### PR TITLE
Update docs to cover CBMC submodule, unit_tests and regression tests

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

File: testsuite/gnat2goto/README.md [hunk 1]
```diff
@@ -47,6 +47,26 @@ See also:
 * http://llvm.org/docs/CommandGuide/lit.html
 * http://www.airs.com/blog/archives/499
 
+CBMC submodule
+--------------
+
+The internmediate JSON files generated by gnat2goto are strongly coupled to
+the particular version of CBMC they are targeting. While they may work with
+a differnet version of CBMC, there is no guarantee that this will be the case.
```
Comment (xbauch) [inline testsuite/gnat2goto/README.md]: different

Comment (chrisr-diffblue) [reply testsuite/gnat2goto/README.md]: Fixed, thanks.

File: testsuite/gnat2goto/README.md [hunk 2]
```diff
@@ -47,6 +47,26 @@ See also:
 * http://llvm.org/docs/CommandGuide/lit.html
 * http://www.airs.com/blog/archives/499
 
+CBMC submodule
+--------------
+
+The internmediate JSON files generated by gnat2goto are strongly coupled to
```
Comment (NlightNFotis) [inline testsuite/gnat2goto/README.md]: *intern*mediate -> `intermediate`

File: gnat2goto/COMPILING.md [hunk 3]
```diff
@@ -47,3 +47,44 @@ Assuming that you've built the dependencies as per the above instructions
...
 
 ` $ install/bin/gnat2goto examples/foo.adb`
+
+# Building and running gnat2goto unit tests
+
+gnat2goto includes a selection of unit tests. These are built as a seperate
+`unit_tests` binary. Assuming you have built gnat2goto as per the above
+instructions, the `unit_tests` binary can be built with the following command
+
+`$ gprbuild -P unit_tests.gpr -cargs -largs`
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/COMPILING.md]: What is this `-cargs` and `-largs` stuff? I've never used those when building the unit tests (also worth pointing out that you don't need to rebuild gnat2goto all the time, you only need to invoke the `Makefile` again if you changed the irep specs)

Comment (chrisr-diffblue) [reply gnat2goto/COMPILING.md]: For the -cargs and -largs - don't honestly know, but thats what is being run in CI.

File: testsuite/gnat2goto/README.md (hunk 2 above)
Comment (chrisr-diffblue) [reply testsuite/gnat2goto/README.md]: I should use a spell checker.... Fixed, thanks.

**Pull request merged**